pep-0597.rst

```diff
@@ -33,135 +34,140 @@ Using the default encoding is a common mistake
+  (but could change if future versions of Python modify the default)
 
-In this point of view, ``open(filename)`` is not readable.
+From this point of view, ``open(filename)`` is not readable code.
 
 ``encoding=locale.getpreferredencoding(False)`` can be used to
-specify the locale encoding explicitly. But it is too long and easy
-to misuse. (e.g. forget to pass ``False`` to its parameter)
+specify the locale encoding explicitly, but it is too long and easy
+to misuse (e.g. one can forget to pass ``False`` as its argument).
 
 This PEP provides an explicit way to specify the locale encoding.
 
 
 Prepare to change the default encoding to UTF-8
 -----------------------------------------------
 
-Since UTF-8 becomes de-facto standard text encoding, we might change
-the default text encoding to UTF-8 in the future.
+Since UTF-8 has become the de-facto standard text encoding,
+we might default to it for opening files in the future.
 
-But this change will affect many applications and libraries. If we
-start emitting ``DeprecationWarning`` everywhere ``encoding`` option
-is omitted, it will be too noisy and painful.
+However, such a change will affect many applications and libraries.
+If we start emitting ``DeprecationWarning`` everywhere the ``encoding``
+argument is omitted, it will be too noisy and painful.
 
-Although this PEP doesn't propose to change the default encoding,
-this PEP will help the change:
+Although this PEP doesn't propose changing the default encoding,
+it will help enable that change by:
 
-* Reduce the number of omitted ``encoding`` options in many libraries
-  before we start emitting the ``DeprecationWarning`` by default.
+* Reducing the number of omitted ``encoding`` arguments in libraries
+  before we start emitting a ``DeprecationWarning`` by default.
 
-* Users will be able to use ``encoding="locale"`` option to suppress
-  the warning without dropping Python 3.10 support.
+* Allowing users to pass ``encoding="locale"`` to suppress
+  the warning, starting with Python 3.10.
 
 
 Specification
 =============
 
 ``EncodingWarning``
---------------------
+-------------------
 
-Add a new ``EncodingWarning`` warning class which is a subclass of
-``Warning``. It is used to warn when the ``encoding`` option is
-omitted and the default encoding is locale-specific.
+Add a new ``EncodingWarning`` warning class as a subclass of
+``Warning``. It is emitted when the ``encoding`` argument is omitted and
+the default locale-specific encoding is used.
 
 
 Options to enable the warning
-------------------------------
+-----------------------------
 
-``-X warn_default_encoding`` option and the
+The ``-X warn_default_encoding`` option and the
 ``PYTHONWARNDEFAULTENCODING`` environment variable are added. They
 are used to enable ``EncodingWarning``.
 
-``sys.flags.encoding_warning`` is also added. The flag represents
+``sys.flags.encoding_warning`` is also added. The flag is ``True`` when
```

Comment:
This is correct, right? The literal value of the flag is `True` (haven't gone over the PR yet)

Reply:
No. Current pull request uses ``1``.

Reply:
Okay, thanks. I assume that's to be consistent with C and the other flags? Suggestion updates it accordingly.

```suggestion
``sys.flags.encoding_warning`` is also added. The flag is ``1`` when
```

Reply:
I intentionally use the term `true` because ``True`` or ``1`` is just implementation details and sys.flags don't guarantee the types in general. We may change the type without any deprecation.

https://docs.python.org/ja/3/library/sys.html#sys.flags

Reply:
```suggestion
``sys.flags.encoding_warning`` is also added. The flag is truthy when
```

Understood, thanks. I revised it to the standard term "truthy" to clarify more explicitly that users can only rely on it being truthy (i.e. `== True`) and don't mistakenly think "true" refers to the `True` literal.

Reply:
"true" is common term in Python doc.
 
```
$ rg -w true | wc -l
     667

$ rg truthy
reference/compound_stmts.rst
588:   * If the guard evaluates as truthy or missing, the ``block`` inside ``case_block`` is
640:   * If the ``guard`` condition evaluates to "truthy", the case block is

library/ast.rst
1263:    is truthy, the body of the case shall be executed. ``body`` contains a list
1264:    of nodes to execute if the guard is truthy.
```

Reply:
```suggestion
``sys.flags.encoding_warning`` is also added. The flag is true when
```

Okay, thanks. Updated accordingly.

Reply:
Applied @methane 's suggestion instead.